java/vector/src/main/java/org/apache/arrow/vector/table/BaseRow.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.table;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Provides row based access to the data held by a {@link Table}.
+ *
+ * <p>
+ * This API is EXPERIMENTAL.
+ */
+public abstract class BaseRow {
+
+  /** The table we're enumerating. */
+  protected final BaseTable table;
+
+  /** the current row number. */
+  protected int rowNumber = -1;
+
+  /**
+   * Returns the standard character set to use for decoding strings. The Arrow format only supports UTF-8.
+   */
+  private final Charset defaultCharacterSet = StandardCharsets.UTF_8;
```

Comment:
It seems this should be a static, since UTF-8 is part of the standard. (I could see there being a non-final attribute for the case of getting strings out of a binary vector, though.)

Reply:
Yes. I could change that. It was created as a field before I learned that only UTF-8 was supported.